### ISSUE TITLE
offcanvas menu and main menu

### ISSUE THREAD

**Issue by alisamii** [gantry/gantry5#534]
In my layout, I have multiple menus, one in the header and one in the navigation sections. I also have the language switcher in the header section.

I have dropped the mobile menu particle in the off canvas section

It adds the menu from the header section to the off canvas, which isn't what I need.

At the very least, I need the menu from the navigation section to be displayed in the off-canvas. Ideally, I would need to be able to configure the off-canvas to display several items/menus in a particular order:

1. display the main menu (from navigation section)
2. display the secondary menu (from the header section)
3. display a horizontal rule
4. display the languages module

Please advise

**Comment by w00fz** [comment]
There isn't a way to move to offcanvas more than 1 menu at this time. However some RC ago we added the ability to let you target what menu you want to go in the Offcanvas, in case you have more than one menu rendering in the page.
In your Layout Manager, open the settings for the Menu you want to show in Offcanvas (in your case the one in Navigation) and select Mobile Target.

**Comment by alisamii** [comment]
In think it's a very useful feature to be able to place multiple items in the off canvas. For example, the language switcher would probably fit perfectly well in the off canvas. And there should be some flexibility in how the menu is populated beyond just selecting the default menu to display off canvas.

Best regards,

Ali

**Comment by alisamii** [comment]
Any chance this can be added to the 5.2 release?

**Comment by mahagr** [comment]
Right now you can workaround the issue by using ```class="hidden-phone"``` etc and by having 2 copies from the particle or module. It may not be the ideal solution, but it should work.

Moving items to offcanvas is JS, so I have no idea what it would take to implement more general solution for this.

**Comment by tamerg** [comment]
Any update on this?

**Comment by w00fz** [comment]
No, nothing yet, this is marked for future release so it's not been looked at.

**Comment by mixahlos** [comment]
Well I am in the same situation.
I have 2 menu particles in 2 sections, and I need an extra menu particle in offcanvas section.

So I tried to do what Mathias suggested, to use the classes `class="hidden-phone"`  and for the one in the offcanvas `class="visible-phone"` 
But when I do that and add a second menu particle in the offcanvas then the default menu in the navigation section loses the Expand on hover functionality. So no dropdowns.
When I disable the 2nd particle menu in the off canvas then the dropdowns appear again.

Any idea if the issue with the dropdown is easy to be solved ?

**Comment by mahagr** [comment]
Quote from @JoomFX in Gitter:

> @mixahlos @mahagr I noticed that issue couple months ago. I was thinking to report it but I forgot.
> The thing is that whatever settings you have for the first menu in the DOM, they are applied to all other menus.
> @mixahlos in your case, the first menu in the DOM (the one in the Offcanvas I guess) has the "Expand on Hover" option unchecked and this is applied to all other menus

**Comment by coolman01** [comment]
Hi
Are we anywhere with this yet? I would like to be able to add other modules, especially the languge switcher, as mention by @alisamii above.

**Comment by N8Solutions** [comment]
@coolman01 
There a couple of ways you can add the language switcher to the offcanvas section.
You can either add a new row to the offcanvas section and add
a module instance and pick the module you want to assign there
or
add a module position to the new row and assign the language switcher module to that position
or
you can edit the menu by going to the menu in Gantry 5 and add a Module instance there and select the language switcher module.

To add to the above, if you add the language switcher via a module instance to the menu it will show up on both the desktop and mobile section in the offcanvas without having to to anything else. You will also need to add some custom styling targeted at the language switcher module to get it to show nicely in that position.

If you choose to add the language switcher by adding it to the offcanvas section directly then you will also need to add the language switcher to another position on your website so it is visible for desktop and tablet as the offcanvas section won't show up on the desktop unless you tell it to and even then hiding the language switcher in the offcanvas section on desktop and tablet is not, in my opinion, a good idea for a good visitor experience on your site.

@mahagr & @w00fz 
It would still be really nice if we had the ability to add more than 1 menu to the offcanvas section. I have a use case for this where I would need to have 2 visible in the off canvas section but only for certain pages.

**Comment by coolman01** [comment]
@N8Solutions  - thanks for the tips - much appreciated. I've already tried everyone of the methods you mention. When I do manage to get it to show, it promptly disappears with the next page refresh! :-)

**Comment by N8Solutions** [comment]
@coolman01 if you are unable to get it to work either you are doing it incorrectly or you have a cache that isn't clearing. Are you using a CDN like Cloudflare by chance? If you are, you need to make sure you have omitted the administrator section from being cached as that will interfere with saving the settings on the admin side. Also, you will need to remember to clear the CDN cache or disable it while you are still testing the website.

**Comment by hjelperne** [comment]
I have 2 menus I wish to use in offcanvas.
I partially solved it by adding menu item aliases to the main menu that I added the «visible-phone» classes to.
The only issue now is that the gantry menu dosent seem to support alias menu items correctly: they (the alias menu items) are not marked with «active» class in their markup (surrounding li tag) when they are selected.
So near - yet so far (it seems).
I suppose this is a bug in the gantry menu system - as the «active» class is added to both the original and alias menu items when using pure joomla menus.
Hope someone can resolve this - as I think it is (would be) a quite nice workaround - until multiple offcanvas menus are supported. 
If the aliases would only display as selected when they are in fact selected...
